### PR TITLE
WIP: bootkube: move to cluster-bootstrap script command

### PR DESCRIPTION
**Pull request by sttts**

### CODE REVIEW

File: data/data/bootstrap/files/usr/local/bin/bootkube.sh.template [hunk 1]
```diff
@@ -204,13 +115,10 @@ rm --force /etc/kubernetes/manifests/machineconfigoperator-bootstrap-pod.yaml
...
-	--rm \
-	--volume "$PWD:/assets:z" \
-	--volume /etc/kubernetes:/etc/kubernetes:z \
-	--network=host \
-	"${CLUSTER_BOOTSTRAP_IMAGE}" \
-	start --asset-dir=/assets --required-pods openshift-kube-apiserver/openshift-kube-apiserver,openshift-kube-scheduler/openshift-kube-scheduler,openshift-kube-controller-manager/openshift-kube-controller-manager,openshift-cluster-version/cluster-version-operator
+podman run --rm "${CLUSTER_BOOTSTRAP_IMAGE}" script \
+    --release-image-digest=${release} \
+    --etcd-server-urls={{.EtcdCluster}} \
+    --intput-assets-dir=/assets | bash -e
```
Comment (abhinavdahiya) [inline data/data/bootstrap/files/usr/local/bin/bootkube.sh.template]: hmm, what does `| bash -e` do ?

File: data/data/bootstrap/files/usr/local/bin/bootkube.sh.template [hunk 2]
```diff
@@ -204,13 +115,10 @@ rm --force /etc/kubernetes/manifests/machineconfigoperator-bootstrap-pod.yaml
 echo "Starting cluster-bootstrap..."
 
 # shellcheck disable=SC2154
```
Comment (abhinavdahiya) [inline data/data/bootstrap/files/usr/local/bin/bootkube.sh.template]: @sttts can you add some comments here like
* what are the input directories that this `cluster-bootstrap` acts on.
* what are some exit criterias
* also add docs on how to troubleshoot errors from this command (maybe just pointing to docs in the repo `cluster-bootstrap`)

File: data/data/bootstrap/files/usr/local/bin/bootkube.sh.template [hunk 3]
```diff
@@ -21,97 +21,8 @@ MACHINE_CONFIG_CONTROLLER_IMAGE=$(podman run --rm ${release} image machine-confi
...
-KUBE_SCHEDULER_OPERATOR_IMAGE=$(podman run --rm ${release} image cluster-kube-scheduler-operator)
-
-OPENSHIFT_HYPERSHIFT_IMAGE=$(podman run --rm ${release} image hypershift)
-OPENSHIFT_HYPERKUBE_IMAGE=$(podman run --rm ${release} image hyperkube)
-
-CLUSTER_BOOTSTRAP_IMAGE=$(podman run --rm ${release} image cluster-bootstrap)
-
-mkdir --parents ./{bootstrap-manifests,manifests}
-
-if [ ! -d cvo-bootstrap ]
```
Comment (abhinavdahiya) [inline data/data/bootstrap/files/usr/local/bin/bootkube.sh.template]: are you sure you want to include `cvo` in `cluster-bootstrap script` ?

Comment (sttts) [reply data/data/bootstrap/files/usr/local/bin/bootkube.sh.template]: It's part of the pivot from bootstrap static pods to the real masters (in other words: bootkube start was waiting for it). If we find a way to exclude it from that step, we can move it back here.

File: data/data/bootstrap/files/usr/local/bin/bootkube.sh.template (hunk 2 above)
Comment (sttts) [reply data/data/bootstrap/files/usr/local/bin/bootkube.sh.template]: on it.

**Comment by crawford** [comment]
This looks to be abandoned. Please rebase and reopen if you are still interested in this functionality.

/close

**Comment by sttts** [comment]
/reopen

It's not abandoned.

**Comment by crawford** [comment]
Looks like this needs a rebase.

**Comment by abhinavdahiya** [comment]
closing due to inactivity. Please reopen if needed.

/close

**Pull request closed without merging**